contrib/opencensus-ext-flask/opencensus/ext/flask/flask_middleware.py

```diff
@@ -142,11 +142,11 @@ def _before_request(self):
             # Set the span name as the name of the current module name
             span.name = '[{}]{}'.format(
                 flask.request.method,
-                flask.request.url)
+                flask.request.base_url)
             tracer.add_attribute_to_current_span(
                 HTTP_METHOD, flask.request.method)
             tracer.add_attribute_to_current_span(
-                HTTP_URL, str(flask.request.url))
+                HTTP_URL, str(flask.request.base_url))
```

Comment:
Using `method + " " + base_url` as span name looks good.

Using `base_url` in HTTP span attribute is **against** the [specification](https://github.com/census-instrumentation/opencensus-specs/blob/master/trace/HTTP.md#attributes).
I suggest creating a GitHub issue in the [spec repo](https://github.com/census-instrumentation/opencensus-specs) and send PR to update the specification, then implement the spec in Python SDK.

Reply:
Thanks, I will update the code it will follow the specification.

Reply:
Awesome! FYI we're also open to making the spec better (including privacy & security concerns), any recommendations/ideas are very welcomed.